src/components/Switch.tsx

```diff
@@ -26,21 +26,30 @@ type SwitchProps = {
 
     /** Callback to fire when the switch is toggled in disabled state */
     disabledAction?: () => void;
+
+    /**
+     * Callback function triggered only after a successful change
+     * in the external state of the switch, after the switch state change animation is triggered
+     */
+    onStateChange?: (isOn: boolean) => void;
 };
 
 const OFFSET_X = {
     OFF: 0,
     ON: 20,
 };
 
-function Switch({isOn, onToggle, accessibilityLabel, disabled, showLockIcon, disabledAction}: SwitchProps) {
+function Switch({isOn, onToggle, accessibilityLabel, disabled, showLockIcon, disabledAction, onStateChange}: SwitchProps) {
     const styles = useThemeStyles();
     const offsetX = useSharedValue(isOn ? OFFSET_X.ON : OFFSET_X.OFF);
     const theme = useTheme();
 
     useEffect(() => {
         offsetX.set(withTiming(isOn ? OFFSET_X.ON : OFFSET_X.OFF, {duration: 300}));
-    }, [isOn, offsetX]);
+        if (onStateChange) {
+            onStateChange(isOn);
```

Comment:
Can't we use `onToggle` for this? 🤔

Reply:
Unfortunately it seems to me that it does not, because doing it in onToggle called the animation trigger at the wrong time, but I will make sure that in the final version of the code it did not become possible

Reply:
I was able to remove the use of this method and it is no longer needed.